integrations/philip-click-up/click_up/client.py

```diff
@@ -0,0 +1,116 @@
+        self.client.timeout = Timeout(30)
+
+    def milesecond_time_to_date_str(self, data: dict):
+
+        if data.get("date_created"):
+            date_created_ms = int(data["date_created"]) / 1000
+
+            data["date_created"] = datetime.fromtimestamp(timestamp=date_created_ms)
+
+        if data.get("date_updated"):
+            date_updated_ms = int(data["date_updated"]) / 1000
+
+            data["date_updated"] = datetime.fromtimestamp(timestamp=date_updated_ms)
+
+        return data
+
+    async def __generate_teams(self):
+
+        response = await self.client.get(url=f"{self.click_up_api_url}/team")
+
+        response.raise_for_status()
+
+        return response.json()
+
+    async def __get_spaces(self, team_id: int):
+
+        response = await self.client.get(url=f"{self.click_up_api_url}/{team_id}/space")
+
+        response.raise_for_status()
+
+        return response.json()
+
+    async def __get_folder(self, space_id: str):
+
+        response = await self.client.get(
+            url=f"{self.click_up_api_url}/space/{space_id}/folder"
+        )
+
+        response.raise_for_status()
+
+        return response.json()
+
+    async def __get_projects(self, folder_id: int):
+        response = await self.client.get(
+            url=f"{self.click_up_api_url}/folder/{folder_id}/list"
+        )
+
+        response.raise_for_status()
+
+        return response.json()
+
+    async def __get_issues(self, list_id: int):
+
+        response = await self.client.get(url=f"{self.click_up_api_url}/{list_id}/space")
+
+        response.raise_for_status()
+
+        return response.json()
+
+    async def get_single_issue(self, issue_id: str):
+
+        response = await self.client.get(url=f"{self.click_up_api_url}/task/{issue_id}")
+
+        response.raise_for_status()
+
+        return response.json()
+
+    async def get_single_project(self, project_id: int):
+        response = await self.client.get(
+            url=f"{self.click_up_api_url}/list/{project_id}"
+        )
+
+        response.raise_for_status()
+
+        return response.json()
+
+    async def get_teams(self) -> AsyncGenerator[list[dict[str, Any]], None]:
+
+        teams = await self.__generate_teams()["teams"]
+
+        yield teams
+
+    async def get_projects(self) -> AsyncGenerator[list[dict[str, Any]], None]:
+        async for team in await self.get_teams():
+            spaces = await self.__get_spaces(team_id=team["id"])["spaces"]
+
+            for space in spaces:
+                folders = await self.__get_folder(space_id=space["id"])["folders"]
+
+                yield folders["lists"]
+
+    async def get_issues(self) -> AsyncGenerator[list[dict[str, Any]], None]:
+        async for project in self.get_projects():
+
+            tasks = await self.__get_issues(list_id=project["id"])["tasks"]
+            tasks = list(
+                map(lambda data: self.milesecond_time_to_date_str(data=data), tasks)
+            )
+            yield tasks
```

Comment:
The issues endpoint supports pagination but I don't see that implemented in the code

Reply:
Done.